source/nodejs/adaptivecards-ui-testapp/src/rendering-utils.ts

```diff
@@ -63,25 +63,33 @@ export function renderCard(cardJson: any, callbackFunction: Function): void {
     // Set the adaptive card's event handlers. onExecuteAction is invoked
     // whenever an action is clicked in the card
     adaptiveCard.onExecuteAction = (action: Action) => {
-        const card: AdaptiveCard = adaptiveCard;
-        const inputs: Input[] = card.getAllInputs(true);
-
+        const actionType: string = action.getJsonTypeName();
         let inputsAsJson: string = "";
-        let isFirstInput: boolean = true;
 
-        inputs.forEach((input) => {
-            const inputId: string = input.id;
-            const inputValue: string = input.value;
+        if (actionType === SubmitAction.JsonTypeName || actionType === ExecuteAction.JsonTypeName){
+            const card: AdaptiveCard = adaptiveCard;
+            const inputs: Input[] = card.getAllInputs(true);
+
+            let isFirstInput: boolean = true;
+
+            inputs.forEach((input) => {
+                const inputId: string = input.id;
+                const inputValue: string = input.value;
 
-            if (!isFirstInput) {
-                inputsAsJson = inputsAsJson.concat(",");
-            }
+                if (!isFirstInput) {
+                    inputsAsJson = inputsAsJson.concat(",");
```

Comment:
instead of building up a JSON string by hand, you should be able to do something like this:

```typescript
const inputs = card.getAllInputs(true);
const inputsObject = {};
inputs.forEach((input) => {
    inputsObject[input.id] = input.value; // beware: last writer wins
});

// later, when you need the string
const inputsAsJsonString = JSON.stringify(inputsObject);
```

Reply:
note: you don't have to do this -- just thought it might be helpful :)